koku/masu/database/report_manifest_db_accessor.py

```diff
@@ -146,3 +146,31 @@ def get_last_seen_manifest_ids(self, bill_date):
             if not processed:
                 assembly_ids.append(manifest.assembly_id)
         return assembly_ids
+
+    def delete_cost_usage_reports_older_than(self, provider_type, billing_period_start_datetime):
+        """
+        Deletes Cost usage Reports older than expiration_date
+
+        Args:
+            provider_type   (String) the provider type to delete
+            billing_period_start_datetime (datetime.datetime) delete all manifests older than this date, exclusive.
+        """
+        recordsToDelete = CostUsageReportManifest.objects.filter(
+            provider__type=provider_type, billing_period_start_datetime__lt=billing_period_start_datetime
+        ).count()
+        CostUsageReportManifest.objects.filter(
+            provider__type=provider_type, billing_period_start_datetime__lt=billing_period_start_datetime
+        ).delete()
```

Comment:
When you call `.delete()` it will return the number of items it deletes. I would take a look at what they do here:  https://github.com/project-koku/koku/blob/master/koku/masu/processor/azure/azure_report_db_cleaner.py#L79

The way simulate was implemented in the past is that they would have a general log that would be called regardless of the deletion occurred that would highlight what vars were used in the query. So like:
```
LOG.info(
                        "Cost usage report manifest removed for provider type: %s with billing period: %s",
                        provider_type,
                        billing_period_start_datetime,
                    )
```
And then the count was only used in the log if the delete actually occurred. 
```
if not simulate:
     delete_count = CostUsageReportManifest.objects.filter(
            provider__type=provider_type, billing_period_start_datetime__lt=billing_period_start_datetime
        ).delete()
        LOG.info("Removing %s cost usage items for billing period start %s", del_count, bill_info)
```

Reply:
@myersCody added

Reply:
@myersCody changed